tests/testing_tools.py

```diff
@@ -53,3 +51,13 @@ def create_test_images(self, barcode_list: list, date_string: str, color: str):
             os.makedirs(os.path.dirname(expected_image_path), exist_ok=True)
             print(f"Created directory: {os.path.dirname(expected_image_path)}")
             image.save(expected_image_path)
+
+    def generate_random_md5(self):
```

Comment:
new function to generate random md5 hashes for naming test directories for unittests to avoid overlapping with and overwriting real directories in program file.